packages/shared/src/components/fields/Checkbox.tsx

```diff
@@ -30,13 +31,14 @@ export const Checkbox = forwardRef(function Checkbox(
     className,
     checkmarkClassName,
     onToggleCallback,
-    id,
+    id = '',
     disabled,
     ...props
   }: CheckboxProps,
   ref: LegacyRef<HTMLInputElement>,
 ): ReactElement {
   const [actualChecked, setActualChecked] = useState(checked);
+  const checkId = useId();
```

Comment:
Make sure to have a valid id on each use, sometime right now we do not have id and as a consequence